src/frontend/js/components/PaymentButton/index.spec.tsx

```diff
@@ -789,6 +789,11 @@ describe.each([
       fetchMock.resetHistory();
       // - Wait until order has been polled 29 times.
       await jest.advanceTimersToNextTimerAsync(PAYMENT_SETTINGS.pollLimit);
+      await act(async () => {
+        jest.runOnlyPendingTimers();
+      });
+
+      // FIXME(rlecellier): test fail here
```

Comment:
After some search, I think the best way to fix this test is the following : 

```ts
// - Wait until order has been polled 29 times.
await act(async () => {
  await jest.advanceTimersByTimeAsync(
    (PAYMENT_SETTINGS.pollLimit - 1) * PAYMENT_SETTINGS.pollInterval,
  );
});
await waitFor(async () => {
  expect(fetchMock.calls()).toHaveLength(PAYMENT_SETTINGS.pollLimit - 1);
});
```

Using `jest.advanceTimersByTimeAsync` looks better as the advance of the timer match the expecting fetch calls count.

Furthermore, I wrapped the timer advance logic into an `act` in order to remove an act warning.